src/pages/recipes/[recipeId].tsx

```diff
@@ -1,14 +1,147 @@
+              <div>
+                Nationalities:
+                <div className="container w-40 overflow-scroll">
+                  {data!.nationalities.length == 0 ? (
+                    <div>No nationalities</div>
+                  ) : (
+                    <div className="h-14">
+                      {data!.nationalities.map((id, key) => (
+                        <div key={key}> {id.nationalityId} </div>
+                      ))}
+                    </div>
+                  )}
+                </div>
+              </div>
+              <div>
+                Cooking methods:
+                <div className="container w-40 overflow-scroll">
+                  {data!.nationalities.length == 0 ? (
+                    <div>No cooking methods</div>
+                  ) : (
+                    <div className="h-14">
+                      {data!.cookingMethods.map((id, key) => (
+                        <div key={key}> {id.cookingMethodId} </div>
+                      ))}
+                    </div>
+                  )}
+                </div>
+              </div>
+              <div>
+                Meal types:
+                <div className="container w-40 overflow-scroll">
+                  {data!.mealTypes.length == 0 ? (
+                    <div>No meal types</div>
+                  ) : (
+                    <div className="h-14">
+                      {data!.mealTypes.map((id, key) => (
+                        <div key={key}> {id.mealTypeId} </div>
+                      ))}
+                    </div>
+                  )}
+                </div>
+              </div>
+            </div>
+          </div>
+        </div>
+        <div className="text-2xl font-bold">
+          Ingredients required to make {data!.name}
+        </div>
+        <div>
+          Make sure you have all of these ingredients to successfully make{" "}
+          {data!.name}!
+        </div>
+        <div className="flex">
+          <ul className="list-disc bg-amber-100 p-5 pl-5">
+            {data!.ingredients.map((id, key) => (
+              <li key={key}>{id.name}</li>
+            ))}
+          </ul>
+        </div>
+        <div className="text-2xl font-bold">How to make {data!.name}</div>
+        <div>
+          Below you can find the full step-by-step instructions to make{" "}
+          {data!.name}! We suggest following the steps to a tee!
+        </div>
+        <div className="flex">
+          <ul className="list-decimal bg-amber-100 p-5 pl-6">
+            {data!.steps.map((id, key) => (
+              <li key={key}>{id.name}</li>
+            ))}
+          </ul>
+        </div>
+        <div className="flex flex-col gap-2">
+          <div className="text-xl">
+            Finished? We hope you enjoy this recipe for {data!.name}!
+          </div>
+        </div>
+      </div>
+    </div>
+  );
+
+  async function copy() {
+    const el = document.createElement("input");
```

Comment:
What's the input for? Can you do await navigator.clipboard.writeText(window.location.href) instead?

Reply:
Yup, I can and it works. I am using this instead of the previous version.